File: PRID5853-G06/AngularClient/ClientApp/src/app/components/models/post.ts
import {  Tag  } from "../models/Tag";
import {  Author  } from "../models/Author";
import {  Vote  } from "../models/Vote";
import { Comment } from "../models/comment";

export class Post {
    id: number;
    title: string;
    body: string;
    timestamp: Date;
    editTimeStamp : Date;
    replies: Post[] = [];
    nbReplies : number;
    acceptedAnswerId: number;
    author: Author;
    comments: Comment[] = [];
    score: number;
    lastActivityTimeStamp : Date;
    tags: Tag[] = [];
    votes: Vote[];
    nbViews: number;
    parentId : number;

    constructor(id = 0, title = "", body = "", timestamp = new Date(), editTimeStamp = null, replies = [], nbReplies = 0, acceptedAnswerId = null, author = null, comments = [], score = 0, nbViews = 0, lastActivityTimeStamp = null, tags = [], votes = [], parentId = 0){
        this.id = id;
        this.title = title;
        this.body = body;
        this.timestamp = timestamp;
        this.editTimeStamp = editTimeStamp;
        this.replies = replies;
        this.nbReplies = nbReplies;
        this.acceptedAnswerId = acceptedAnswerId;
        this.author = author;
        this.comments = comments;
        this.score = score;
        this.nbViews = nbViews;
        this.lastActivityTimeStamp = lastActivityTimeStamp;
        this.tags = tags;
        this.votes = votes;
        this.parentId = parentId;
    }
    

}

File: PRID5853.G06.Entities/Models/Post.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace PRID5853.G06.Entities.Models
{
    public class Post : EntityBase
    {
        public string Title { get; set; }
        [Required]
        public string Body { get; set; }
        [Required]
        public DateTime Timestamp { get; set; }
        public DateTime? EditTimeStamp { get; set; }

        public virtual ICollection<Comment> Comments { get; set; }
        public virtual User Author { get; set; }
        public int AuthorId { get; set; }
        public virtual ICollection<Post> Replies { get; set; }
        public virtual ICollection<Vote> Votes { get; set; }
        public virtual ICollection<PostTag> PostTags { get; set; }
        public virtual Post Parent { get; set; }
        public int? ParentId { get; set; }
        public int? AcceptedAnswerId { get; set; }
        public int NbViews { get; set; } = 0; // Implementing a basic counter (If we have time - Make it better by counting only distinct users).
        [NotMapped]
        public IEnumerable<Tag> Tags { get => PostTags?.Select(pt => pt.Tag);}
    }
}
